File: BozukProjeler/YKKEntegrasyonVB-eski/Ykk.WebApi.ClientConsole/ClientExtended.vb
Imports System.Net.Http.Headers

Namespace Ykk.WebApi.ClientConsole
    Partial Public Class Client
        Public Property Username() As String
        Public Property Password() As String

        Private Sub PrepareRequest(ByVal client As System.Net.Http.HttpClient, ByVal request As System.Net.Http.HttpRequestMessage, ByVal url As String)
            If request.RequestUri.AbsolutePath.Contains("Authentication") Then
                Return
            End If
            Dim token = Me.LoginAsync(New LoginModel With {
                .Password = Me.Password,
                .Username = Me.Username
            }).Result
            request.Headers.Authorization = New AuthenticationHeaderValue("Bearer", token)
        End Sub
    End Class
End Namespace

File: BozukProjeler/YKKEntegrasyonCSharp-eski/ClientExtended.cs
using System.Net.Http.Headers;

namespace YKKEntegrasyon
{
    public partial class Client
    {
        public string Username { get; set; }
        public string Password { get; set; }

        partial void PrepareRequest(System.Net.Http.HttpClient client, System.Net.Http.HttpRequestMessage request,
            string url)
        {
            if (request.RequestUri.AbsolutePath.Contains("Authentication")) return;
            var token = this.LoginAsync(new LoginModel
            {
                Password = this.Password,
                Username = this.Username
            }).Result;
            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
        }
    }
}
